src/target_triple.rs

```diff
@@ -0,0 +1,97 @@
+use crate::component;
+use anyhow::{bail, Result};
+use std::fmt;
+
+#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct TargetTriple(String);
+
+impl fmt::Display for TargetTriple {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl TargetTriple {
+    pub fn new(name: &str) -> Self {
```

Comment:
Removed this altogether as well, since the only place of usage was in `name_allowed` for comparison (which we can use `to_string()` to compare anyway)